packages/aws-cdk-lib/core/test/stack.test.ts

```diff
@@ -39,7 +39,49 @@ describe('stack', () => {
     }).toThrow(`Stack name must be <= 128 characters. Stack name: '${reallyLongStackName}'`);
   });
 
-  test('stack objects have some template-level propeties, such as Description, Version, Transform', () => {
+  test.each([
+    ['Has:Colon', 'HasColon'],
+    ['0startWithNumber', 'startWithNumber'],
+    ['Has-Dash', 'Has-Dash'],
+    [undefined, 'Default'],
+    ['With_Underscore', 'WithUnderscore'],
+    ['with.dot', 'withdot'],
+    ['with/slash', 'with--slash'],
+    ['with space', 'withspace'],
+    ['UPPERCASE', 'UPPERCASE'],
+    ['mixedCase123', 'mixedCase123'],
+    ['!@#$%^', 'Default'],
+    ['123456', 'Default'],
+    ['a-b-c', 'a-b-c'],
+    ['x_y_z', 'xyz'],
+    ['abc.def.ghi', 'abcdefghi'],
```

Comment:
All these are additional ways to provide IDs that might be invalid as a stack name.